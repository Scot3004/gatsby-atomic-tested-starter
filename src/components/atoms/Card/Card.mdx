---
name: Card
---

import { Playground } from 'docz';
import Card from '.';
import CardContent from '../CardContent';

# Card

Renders an Card


## Playground

<Playground>
  <Card>
    <CardContent>
      Here should be a card content
    </CardContent>
  </Card>

  <Card>
    but is usable without card content
  </Card>
</Playground>
