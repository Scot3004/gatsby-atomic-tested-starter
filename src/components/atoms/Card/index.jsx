import styled from 'styled-components';

export const Card = styled.div`
  align-items: center;
  border-radius: 0.5rem;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  box-sizing: border-box;
  display: flex;
  margin: 0 0 1rem 0;
  padding: 0.6rem;
  transition: 0.3s;
  width: 100%;

  &:last-child {
    margin: 0;
  }
`;

export default Card;
