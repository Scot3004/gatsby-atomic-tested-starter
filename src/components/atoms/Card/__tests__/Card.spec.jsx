import React from 'react'
import 'jest-styled-components'
import renderer from 'react-test-renderer'
import Card from '..'

test('it works', () => {
  const tree = renderer.create(<Card />).toJSON()
  expect(tree).toMatchSnapshot()
})
