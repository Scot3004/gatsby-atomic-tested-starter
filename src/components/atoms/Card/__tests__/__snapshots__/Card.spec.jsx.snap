// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I want to see a card element Then the card should have borders and a good spacing 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0 1rem 0;
  padding: 0.6rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  width: 100%;
}

<div
  className="c0"
/>
`;
